soapboxd/activity.go

```diff
@@ -0,0 +1,182 @@
+package soapboxd
+
+import (
+	"database/sql"
+	"time"
+
+	"github.com/adhocteam/soapbox/models"
+	"github.com/adhocteam/soapbox/proto"
+	pb "github.com/adhocteam/soapbox/proto"
+	gpb "github.com/golang/protobuf/ptypes/timestamp"
+	"github.com/pkg/errors"
+	"golang.org/x/net/context"
+)
+
+var activityTypeToModel = map[pb.ActivityType]models.ActivityType{
+	pb.ActivityType_APPLICATION_CREATED:   models.ActivityTypeApplicationCreated,
+	pb.ActivityType_DEPLOYMENT_STARTED:    models.ActivityTypeDeploymentStarted,
+	pb.ActivityType_DEPLOYMENT_SUCCESS:    models.ActivityTypeDeploymentSuccess,
+	pb.ActivityType_DEPLOYMENT_FAILURE:    models.ActivityTypeDeploymentFailure,
+	pb.ActivityType_ENVIRONMENT_CREATED:   models.ActivityTypeEnvironmentCreated,
+	pb.ActivityType_ENVIRONMENT_DESTROYED: models.ActivityTypeEnvironmentDestroyed,
+}
+
+func (s *server) AddActivity(ctx context.Context, activity *pb.Activity) (*pb.Empty, error) {
+	userID := activity.UserId
+	applicationID := activity.ApplicationId
+	deploymentID := activity.DeploymentId
+	environmentID := activity.EnvironmentId
+
```

Comment:
You could avoid the temporary stack variables and tidy it up somewhat by grouping these into a `interface{}` slice and using `...` in the call to `Exec`:

`params := []interface{`,

`s.db.Exec(query, ...params)`